### PR TITLE
Render image fills on the canvas

### PR DESCRIPTION
**Pull request by milg15**
Render Pattern preview

### CODE REVIEW

File: packages/noya-renderer/src/primitives.ts [hunk 1]
```diff
@@ -163,8 +165,120 @@ export function fill(
...
+                CanvasKit.Matrix.scaled(scale, scale),
+              ),
+            ),
+          );
+          break;
+        }
+        case Sketch.PatternFillType.Stretch: {
+          paint.setShader(
+            canvasImage.makeShaderCubic(
+              CanvasKit.TileMode.Clamp,
```
Comment (dabbott) [inline packages/noya-renderer/src/primitives.ts:230]: I noticed the other 2 are `Decal` and this one is `Clamp` - is it different in this case? If not, let's use `Decal` just for consistency

File: packages/noya-renderer/src/primitives.ts [hunk 2]
```diff
@@ -163,8 +165,120 @@ export function fill(
...
+          const scale = Math.max(
+            scaledRect.width / canvasImage.width(),
+            scaledRect.height / canvasImage.height(),
+          );
+
+          paint.setShader(
+            canvasImage.makeShaderCubic(
+              CanvasKit.TileMode.Decal,
+              CanvasKit.TileMode.Decal,
+              1,
```
Comment (dabbott) [inline packages/noya-renderer/src/primitives.ts:214]: Does the 1 help here?

**Review by dabbott — APPROVED** [review]
Looks great! 

I think after this we can refactor some of the scaling stuff into the `resize` util so it's easier to re-use. I didn't realize that we'd pretty much always need to know the scale when we use `resize`. After that, I think we can combine all 3 cases, fill/fit/stretch, into one.

File: packages/noya-renderer/src/primitives.ts (hunk 2 above)
Comment (milg15) [reply packages/noya-renderer/src/primitives.ts:214]: I am not sure, I'll leave it to 0 for consistency.

**Pull request merged**